fabric-common-tags-api-v1/src/main/java/net/fabricmc/fabric/api/tag/v1/CommonItemTags.java

```diff
@@ -0,0 +1,109 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.tag.v1;
+
+import net.minecraft.item.Item;
+import net.minecraft.tag.TagKey;
+
+import net.fabricmc.fabric.impl.tag.common.TagRegistration;
+
+/**
+ * See {@link net.minecraft.tag.ItemTags} for vanilla tags.
+ * Note that addition to some vanilla tags implies having certain functionality.
+ */
+public class CommonItemTags {
+	// Tool tags
+	public static final TagKey<Item> PICKAXES = register("pickaxes");
+	public static final TagKey<Item> SHOVELS = register("shovels");
+	public static final TagKey<Item> HOES = register("hoes");
+	public static final TagKey<Item> AXES = register("axes");
+	public static final TagKey<Item> SHEARS = register("shears");
+	/**
+	 * For throwable weapons, like Minecraft tridents.
+	 */
+	public static final TagKey<Item> SPEARS = register("spears");
+	public static final TagKey<Item> SWORDS = register("swords");
+	public static final TagKey<Item> BOWS = register("bows");
+	// Ores and ingots
+	public static final TagKey<Item> IRON_INGOTS = register("iron_ingots");
+	public static final TagKey<Item> RAW_IRON_ORES = register("raw_iron_ores");
+	public static final TagKey<Item> RAW_IRON_BLOCKS = register("raw_iron_blocks");
+	public static final TagKey<Item> IRON_ORES = register("iron_ores");
+	public static final TagKey<Item> GOLD_ORES = register("gold_ores");
+	public static final TagKey<Item> RAW_GOLD_ORES = register("raw_gold_ores");
+	public static final TagKey<Item> RAW_GOLD_BLOCKS = register("raw_gold_blocks");
+	public static final TagKey<Item> GOLD_INGOTS = register("gold_ingots");
+	public static final TagKey<Item> REDSTONE_DUSTS = register("redstone_dusts");
+	public static final TagKey<Item> REDSTONE_ORES = register("redstone_ores");
+	public static final TagKey<Item> COPPER_INGOTS = register("copper_ingots");
+	public static final TagKey<Item> RAW_COPPER_ORES = register("raw_copper_ores");
+	public static final TagKey<Item> RAW_COPPER_BLOCKS = register("raw_copper_blocks");
+	public static final TagKey<Item> COPPER_ORES = register("copper_ores");
+	public static final TagKey<Item> ORES = register("ores");
+	public static final TagKey<Item> NETHERITE_ORES = register("netherite_ores");
+	public static final TagKey<Item> NETHERITE_INGOTS = register("netherite_ingots");
+	public static final TagKey<Item> QUARTZ_ORES = register("quartz_ores");
+	public static final TagKey<Item> QUARTZ = register("quartz");
+	public static final TagKey<Item> LAPIS_ORES = register("lapis_ores");
+	public static final TagKey<Item> LAPIS = register("lapiss");
+	public static final TagKey<Item> DIAMOND_ORES = register("diamond_ores");
+	public static final TagKey<Item> DIAMONDS = register("diamonds");
```

Comment:
I am not too sure about the tags for non-metal resources, since finding a plural form for the material itself can be annoying: `quartzes` or `quartz`, `rubies` is quite different from `ruby+something`, `lapis` is both singular and plural. But I don't really have a better suggestion than case-by-case.

Reply:
The rule has been to use the plural form or when available the bulk noun for tags.